manifests/user/crontab.pp

```diff
@@ -0,0 +1,95 @@
+# == Class: cron::user::crontab
+#
+#  About:  This class is to manage user crontabs.
+#
+#  Purpose:  Often application teams or application users
+#            have crontabs that must execute as the application user.
+#            For example:  DBAs may want to execute crontab entries
+#            as the user oracle, mysql, etc.  Same goes with other
+#            commercial and proprietary applications operating in a
+#            Least Required Permissions model.
+#
+#  Usage:
+#      $user_crontabs => {
+#          'user1' => {'vars' => [ 'SHELL': '/bin/bash', 'MYECHO': '$(which echo)' ], 'entries' => [ '# Echo Hello World': '* 1 * * * $MYECHO "Hello World!"' ]},
+#          'user2' => {'vars' => [ 'SHELL': '/bin/bash', 'MYECHO': '$(which echo)' ], 'entries' => [ '# Echo Hello World': '* 3 * * * $MYECHO "Hello user2!"' ]}
+#      }
+#      create_resources(cron::user::crontab, $user_crontabs)
+#
+#
+#  Hiera data structure to be used by the parent class cron:
+#      cron::user_crontabs:
+#        'user1':
+#          vars:
+#            'SHELL': '/bin/bash'
+#            'MYECHO': '$(which echo)'
+#          entries:
+#            '# Echo Hello World': '* 1 * * * $MYECHO "Hello World!" 2>&1'
+#        'user2':
+#          vars:
+#            'SHELL': '/bin/bash'
+#            'MYECHO': '$(which echo)'
+#          entries:
+#            '# Echo Hello World': '* 3 * * * $MYECHO "Hello user2!" 2>&1'
+#
+#
+#
+define cron::user::crontab (
+  $ensure  = file,
```

Comment:
`ensure` doesn't bring functionallity as it isn't used in the file resource further down.